CodeGen/src/nuttx_main.c

```diff
@@ -92,54 +125,85 @@ static void *rt_task(void *p)
   mlockall(MCL_CURRENT | MCL_FUTURE);
 #endif
 
-  Tsamp = NAME(MODEL,_get_tsamp)();
+  while (!end) {
+    Tsamp = NAME(MODEL,_get_tsamp)();
+
+    t_isr.tv_sec =  0L;
+    t_isr.tv_nsec = (long)(1e9*Tsamp);
+    tsnorm(&t_isr);
 
-  t_isr.tv_sec =  0L;
-  t_isr.tv_nsec = (long)(1e9*Tsamp);
-  tsnorm(&t_isr);
+    T=0;
 
-  T=0;
+    NAME(MODEL,_init)();
+#ifdef CONF_SHV_USED
+    if (!mctx->com_inited) {
+      if (NAME(MODEL, _com_init)(shv_my_at_signlr) == 0) {
+        mctx->com_inited = true;
+      }
+    }
+#endif
+    /* Wait for connections a bit (so the first samples are not delayed) */
 
-  NAME(MODEL,_init)();
+    usleep(1000 * 1000);
 
 #ifdef CANOPEN
-  canopen_synch();
+    canopen_synch();
 #endif
 
-  /* get current time */
-  clock_gettime(CLOCK_MONOTONIC,&t_current);
-  T0 = t_current;
-  
-  while(!end){
+    mctx->running_state = PYSIM_MODEL_CTRLLOOP_RUNNING;
+    puts("CTRLLOOP START");
 
-    /* periodic task */
-    T = calcdiff(t_current,T0);
+    /* get current time */
+    clock_gettime(CLOCK_MONOTONIC,&t_current);
+    T0 = t_current;
     
-    NAME(MODEL,_isr)(T);
+    while (!mctx->ctrlloopend) {
```

Comment:
```suggestion
    while (!mctx->ctrlloopend && !end) {
```

Otherwise linux target can't be terminated by SIGINT.

Reply:
That's true. I'll fix this. I'll let the while condition the same, as I can cause the while loop to end from the SIGINT `endme` callback.

Reply:
Fixed.